### PR TITLE
Patches

### PR DESCRIPTION
**Pull request by MisterDA**
Hello,
This is a bunch of patches for bugs reported by [PVS-Studio](https://www.viva64.com/en/pvs-studio/).

It mostly removes some dead code and checks, and fixes some memory leaks.

PVS-Studio also reports a whole lot of calls to malloc and the family, or strdup calls that are not checked for null results.
Another warning reported are expressions of the form `double x = y / 2` where `y` is of type int, that could be written as `double x = y / 2.` to avoid the loss of a fractional part. This is not fixed in this PR, let me know if it is more-or-less needed.
*Edit: this was fixed in 1372d61: Avoid loss of fractional part*
 

Thanks for the good work!

### CODE REVIEW

File: sway/tree/view.c [hunk 1]
```diff
@@ -709,7 +709,7 @@ void view_update_size(struct sway_view *view, int width, int height) {
 		con->surface_y = fmax(con->surface_y, con->content_y);
 	}
 	con->surface_width = width;
-	con->surface_width = height;
+	con->surface_height = height;
```
Comment (MisterDA) [inline sway/tree/view.c]: Please double-check this. It looks like a copy-mistake to me, but it's possible that I don’t know what I'm doing.

**Comment by MisterDA** [comment]
Hi,
I’ve changed the commit messages with your suggestions.
I also added a commit that divides integers by `2.` (double) when storing the result in a double, to keep a fractional part, although I don’t know if that’s a good idea.

**Comment by progandy** [comment]
> I also added a commit that divides integers by `2.` (double) when storing the result in a double, to keep a fractional part, although I don’t know if that’s a good idea.

There is another pull request that converts the variables to integers #4393

**Comment by MisterDA** [comment]
> > I also added a commit that divides integers by `2.` (double) when storing the result in a double, to keep a fractional part, although I don’t know if that’s a good idea.
> 
> There is another pull request that converts the variables to integers #4393

Ok, so 1372d61 is unneeded. I’m removing it.

File: sway/tree/view.c (hunk 1 above)
Comment (ddevault) [reply sway/tree/view.c]: I agree that this is a copy/paste mistake.

**Comment by ddevault** [comment]
Thanks!

**Pull request merged**